libs/payments/ui/src/lib/nestapp/NextIOValidator.ts

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { plainToInstance } from 'class-transformer';
+import { Validator } from 'class-validator';
+
+/**
+ * Will capture timing for this method and send them to StatsD. Requires the target class to have statsd exposed as a public property.
+ * Timings will appear as ClassName_methodName in StatsD.
+ */
+export function NextIOValidator<
```

Comment:
can we change this to something like `MethodTypeValidator`? `IO` is an overloaded term that feels like less of a "clear" fit here

Reply:
The term `type` is overloaded imo, especially in TypeScript, and additionally means something that doesn't impact runtime.